src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -657,6 +649,9 @@ private static IEnumerable<TSource> TakeWhileIterator<TSource>(IEnumerable<TSour
         public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> source, int count)
         {
             if (source == null) throw Error.ArgumentNull("source");
+            if (count < 0) count = 0;
```

Comment:
This simplifies further logic. What I'd like to do here is `if (count <= 0) return source;` but that might raise objections in terms of exposed interfaces.